queries.sql

```diff
@@ -88,6 +88,252 @@ SELECT animals.name FROM animals JOIN owners ON animals.owner_id = owners.id WHE
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT animals.name, vets.name, visits.date_of_visit
+FROM visits
+INNER JOIN animals ON animals.id = visits.animal_id
+INNER JOIN vets ON vets.id = visits.vet_id
+WHERE vets.name = 'Stephanie Mendez'
+AND visits.date_of_visit BETWEEN '2020-04-01' AND '2020-08-30'
+
+-- What animal has the most visits to vets?
+SELECT animals.name, COUNT(visits.date_of_visit) AS visits
+FROM visits
+INNER JOIN animals ON animals.id = visits.animal_id
+INNER JOIN vets ON vets.id = visits.vet_id
+GROUP BY animals.id
+ORDER BY COUNT(visits.date_of_visit) DESC
+LIMIT 1
+
+-- Who was Maisy Smith's first visit?
+SELECT vets.name, animals.name, visits.date_of_visit
+FROM visits
+INNER JOIN animals ON animals.id = visits.animal_id
+INNER JOIN vets ON vets.id = visits.vet_id
+WHERE vets.name = 'Maisy Smith'
+ORDER BY visits.date_of_visit ASC
+LIMIT 1
+
+-- Details for most recent visit: animal information, vet information, and date of visit.
+SELECT visits.date_of_visit, animals.*, vets.*
+FROM visits
+FULL OUTER JOIN animals ON animals.id = visits.animal_id
+FULL OUTER JOIN vets ON vets.id = visits.vet_id
+ORDER BY visits.date_of_visit ASC
+
+-- How many visits were with a vet that did not specialize in that animal's species?
+SELECT COUNT(*) AS visits_of_vets_not_specialize
+FROM visits
+INNER JOIN vets ON vets.id = visits.vet_id
+INNER JOIN animals ON animals.id = visits.animal_id
+WHERE animals.species_id NOT IN (
+	SELECT coalesce(specializations.spec_id, -1)
+	FROM vets
+	LEFT OUTER JOIN specializations ON specializations.vet_id = vets.id
+	WHERE vets.id = visits.vet_id
+)
+
+-- What specialty should Maisy Smith consider getting? Look for the species she gets the most.
+SELECT vets.name, species.name, COUNT(animals.species_id)
+FROM visits
+INNER JOIN vets ON vets.id = visits.vet_id
+INNER JOIN animals ON animals.id = visits.animal_id
+INNER JOIN species ON species.id = animals.species_id
+WHERE vets.name = 'Maisy Smith'
+GROUP BY vets.name, species.name
+ORDER BY COUNT(animals.species_id) DESC
+LIMIT 1
+
+SELECT * FROM specializations;
+
+-- Create a "join table" called visits
+CREATE TABLE visits (
+animal_id INTEGER,
+vet_id INTEGER,
+date_of_visit DATE,
+CONSTRAINT fk_animals FOREIGN KEY (animal_id) REFERENCES animals(id),
+CONSTRAINT fk_vets FOREIGN KEY (vet_id) REFERENCES vets(id)
+);
+SELECT * FROM visits;
+
+-- Create a "join table" called specializations
+CREATE TABLE specializations (
+spec_id INTEGER,
+vet_id INTEGER,
+CONSTRAINT fk_vets FOREIGN KEY (vet_id) REFERENCES vets(id),
+CONSTRAINT fk_species FOREIGN KEY (spec_id) REFERENCES species(id)
+);
+SELECT * FROM specializations;
+
+-- Insert the following data for visits:
+BEGIN;
+INSERT INTO visits (animal_id, vet_id, date_of_visit)
+SELECT animals.id, vets.id, '2020-05-24' FROM animals, vets
+WHERE animals.name = 'Agumon'
+AND vets.name = 'William Tatcher';
+COMMIT;
+
```

Comment:
- [ ] Please ensure that everything that inserts data into the database is added to the data.sql file and not the queries.sql file. Kindly remove the data insertion lines 👍🏽 . 🙏.